src/bootstrap/setup.rs

```diff
@@ -110,15 +110,50 @@ pub fn setup(src_path: &Path, profile: Profile) {
 
     let build = TargetSelection::from_user(&env!("BUILD_TRIPLE"));
     let stage_path = ["build", build.rustc_target_arg(), "stage1"].join("/");
-    let toolchain_link =
-        Command::new("rustup").args(&["toolchain", "link", "stage1", &stage_path[..]]).output();
-    if toolchain_link.is_err() {
-        println!("WARNING: `rustup`failed to link stage 1 build to `stage1` toolchain");
-        println!(
-            "To manually link stage 1 build to `stage1` tool chain, run:\n
-            `rustup toolchain link stage1 build/{}/stage1`",
-            build.rustc_target_arg()
-        );
+
+    let created = fs::create_dir(&stage_path[..]).is_ok();
+
+    // Link `stage1` toolchain if `rustup` is installed
+    // and we have not already done so.
+    if Path::new("~/.cargo/bin/rustup").exists() && created {
```

Comment:
This is not right. It needs to check if rustup is anywhere in PATH; and ideally if it's executable. Try seeing if `rustup --version` succeeds or something like that.